torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -268,6 +271,19 @@ std::shared_ptr<ReductionParams> innerPersistentHeuristic(
         (int64_t)vectorize_factor);
   }
 
+  // start from small block size to minimize expensive inter-threads reduction
+  const int threads_after_vectorize =
+      inner_most_dimension_numel / inner_reduction_unroll_factor;
+  constexpr int scheduler_per_sm = 4;
+  if (vectorize) {
+    bdimx = std::min(
+        scheduler_per_sm * dev_prop->warpSize, threads_after_vectorize);
+  }
```

Comment:
Doesn't this also affect the outer normalization?

Reply:
Oh, we are not enabling vectorization in the outer case, so this should always be false.

Reply:
Can you please change the condition as follows?
```suggestion
  if (inner_reduction && vectorize) {
    bdimx = std::min(
        scheduler_per_sm * dev_prop->warpSize, threads_after_vectorize);
  }
```